woocommerce/woocommerce
# Adding Stock status to inventory tab and toggle block

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Adds stock status to product block editor inventory section. 

I also ended up refactoring a previous inventory fields block and adding a generic toggle block.

Closes #37884 .


### How to test the changes in this Pull Request:

1. Checkout branch and enable `product-block-editor` feature flag using WCA Test Helper.
2. Navigate to Products -> Add new to see new product block editor.
3. Add a name to the product.
4. Go to Inventory tab.
5. With "track stock quantity" toggle disabled, you should see the "Stock status" radio field.
![image](https://user-images.githubusercontent.com/444632/233477816-92459827-4105-4a3a-b760-dbc2ee5ef6e4.png)

6. Select a different option, refresh, and ensure that your selection persists.
7. Enable the "track stock status" toggle above.
8. The "Stock status" radio should disappear and now display the "available quantity" field.
9. Ensure values for both of those fields also persist when saving and refreshing.
